uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1024.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+trait CRM_HRLeaveAndAbsences_Upgrader_Step_1024 {
+  
+  /**
+   * Update leave request templates to system workflow message
+   *
+   * @return bool
+   */
+  public function upgrade_1024() {
+    $templateName = 'msg_tpl_workflow_leave';
+    $result = $this->up1024_createLeaveWorkflowOptionGroup($templateName);
+    
+    if ($result['count'] > 0) {
+      $this->up1024_createLeaveWorkflowOptionValue($result['id']);
+    }
+    
+    return TRUE;
+  }
+  
+  /**
+   * Create Leave Message Template Workflow Option Group.
+   *
+   * @param $templateName
+   *
+   * @return array
+   */
+  public function up1024_createLeaveWorkflowOptionGroup($templateName) {
+    $workflow = 'Message Template Workflow for Leave';
+    
+    $result = civicrm_api3('OptionGroup', 'create', [
+      'name' => $templateName,
+      'title' => $workflow,
+      'description' => $workflow,
+    ]);
+    CRM_Core_PseudoConstant::flush();
+    
+    return $result;
+  }
+  
+  /**
+   * Create Leave Message Template Workflow Option Values.
+   *
+   * @param $templateName
+   */
+  public function up1024_createLeaveWorkflowOptionValue($templateName) {
+    $optionValues = [
+      [
+        'option_group_id' => $templateName,
+        'label' => 'CiviHR Leave Request Notification',
+        'name' => 'civihr_leave_request',
+      ],
+      [
+        'option_group_id' => $templateName,
+        'label' => 'CiviHR TOIL Request Notification',
+        'name' => 'civihr_toil_request',
+      ],
+      [
+        'option_group_id' => $templateName,
+        'label' => 'CiviHR Sickness Record Notification',
+        'name' => 'civihr_sickness_record',
+      ]
+    ];
+  
+    foreach ($optionValues as $optionValue) {
+      $result = civicrm_api3('OptionValue', 'create', $optionValue);
+      if ($result['count'] > 0) {
+        $this->up1024_updateLeaveMsgTemplate(
+          $result['id'],
+          $optionValue['label']
+        );
+      }
+    }
+  }
+  
+  /**
+   * Update message template workflow id for option group msg_tpl_workflow_leave
+   *
+   * @param $workflowId
+   * @param $msgLabel
+   */
+  public function up1024_updateLeaveMsgTemplate($workflowId, $msgLabel) {
+    civicrm_api3('MessageTemplate', 'get', array(
+      'return' => ['id'],
+      'msg_title' => $msgLabel,
+      'api.MessageTemplate.create' => [
+        'workflow_id' => $workflowId,
+        'id' => '$value.id'
+      ],
+    ));
+  }
```

Comment:
I was checking how System Workflow Message Templates work in CiviCRM and I think your solution will need to be updated a little bit. Here is an example of a few records from the `civicrm_msg_template` table:

![captura de tela de 2018-04-02 12-39-39](https://user-images.githubusercontent.com/388373/38202719-1854a726-3673-11e8-81d3-f4f27feb9c57.png)

As you can see, there are two records for each template. One is marked as default (`is_default` == 1) and the other is marked as reserved (`is_reserved` == 1).

This probably happens because Civi allows you to customize a System Workflow template and then revert any changes you might have done. From a quick investigation I did, it seems that the default template is the one where Civi will save your customizations to and the reserved one works like some sort of "backup", and it is used to get the original template values from in case you want to revert the customizations.

So, if you want to covert these templates to "System Workflow Templates", you'll probably have to follow the same logic. Please make sure to investigate and understand how System Workflow Messages to make sure these templates will be what Civi expect them to be.